pkg/transport/transport.go

```diff
@@ -24,6 +24,13 @@ var (
 	timeout = 20 * time.Second
 )
 
+// Observer implements the ConnSuccess and ConnFailure methods
+// for Discovery and other services to use
+type Observer interface {
+	ConnSuccess(ctx context.Context, node *pb.Node)
```

Comment:
again here you could be explicit and make it ConnectionSuccess - but these are nits so nw